### PR TITLE
Make the flate2 dependency optional

### PR DESCRIPTION
**Pull request by eminence**
CC #156

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -665,13 +665,24 @@ pub enum ConfigSetting {
...
+#[cfg_attr(feature = "flate2", doc = "The flate2 feature is currently enabled")]
+#[cfg_attr(not(feature = "flate2"), doc = "The flate2 feature is NOT currently enabled")]
 pub fn kernel_config() -> ProcResult<HashMap<String, ConfigSetting>> {
-    use flate2::read::GzDecoder;
-
-    let reader: Box<dyn BufRead> = if Path::new(PROC_CONFIG_GZ).exists() {
-        let file = FileWrapper::open(PROC_CONFIG_GZ)?;
-        let decoder = GzDecoder::new(file);
-        Box::new(BufReader::new(decoder))
+    let reader: Box<dyn BufRead> = if Path::new(PROC_CONFIG_GZ).exists() && cfg!(feature = "flate2") {
```
Comment (pymongo) [inline src/lib.rs:675]: I think `&& cfg!(feature = "flate2")` is useless

Comment (pymongo) [reply src/lib.rs:675]: if cfg!(feature = "flate2") == false,  block would not execute

if cfg!(feature = "flate2") == true, block would execute `flate2::read::GzDecoder`

so the cfg(not(...flate2 block may never execute

Comment (pymongo) [reply src/lib.rs:675]: if cfg!(feature = "flate2") == false,  block would not execute

if cfg!(feature = "flate2") == true, block would execute `flate2::read::GzDecoder`

so the cfg(not(...flate2 block may never execute

**Comment by pymongo** [comment]
Other prosible conditional compile logic:

1. if `feature = "flate2"` disable, not compile the pub fn kernel_config function, and all unittest relative it add cfg(feature="flate2") attr
2. like procfs::boot_times(), in chrono feature return Datetime while not chrono feature return i64, I think kernel_config not flate2 feature can return bytes, and flate2 feature return HashMap

File: src/lib.rs (hunk 1 above)
Comment (eminence) [reply src/lib.rs:675]: Yes, the `#[cfg(not(feature = "flate2"))]` block is unreachable, but is needed to typecheck

**Comment by pymongo** [comment]
This PR looks good to me, I can remove the flate2 dependency and speed up my project compile

**Comment by eminence** [comment]
This is a breaking change, since anyone using procfs with `default-features = false` will see an incompatible change in behavior for the `kernel_config` function.  This is unfortunate, but I'm going to live with it.  We'll put a big notice in the release notes about this breaking change.  I'll also looked many crates that depend on procfs and use `default-features = false`, and none use the `kernel_config` function.

**Pull request merged**